[andytoh1] iP

andytoh1 opened the pull request:
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

### Chatbot

Chatbot is: 
- Easy to use
- Helpful
- **Free**

To get started:
1. Download it
2. Run it
3. Use it 😃

Has the following features:
- [x] Add task to list
- [x] View tasks
- [x] Mark tasks as done
- [x] Delete ~~tasks~~ from list 

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

andytoh1 committed:
no message

andytoh1 committed:
no message

andytoh1 committed:
no message

andytoh1 committed:
no message

andytoh1 committed:
no message

andytoh1 committed:
no message

andytoh1 committed:
no message

andytoh1 committed:
no message

andytoh1 committed:
no message

andytoh1 committed:
no message

sopa301 commented on src/main/java/Chatbot.java:
```diff
@@ -0,0 +1,398 @@
+
+    public static class Task {
+        private String description;
+        private boolean done;
```
Perhaps a more boolean-ish name here like isDone would be easier to work with?

sopa301 commented on src/main/java/Chatbot.java:
```diff
@@ -0,0 +1,398 @@
+    }
+
+    public static class Storage {
+        public static void load(TaskList taskList) {
```
I'm curious as to why load is named as such. Is there any reason why you decided to name it like that instead of loadTasks?

sopa301 commented on src/main/java/Chatbot.java:
```diff
@@ -0,0 +1,398 @@
+    public static class Storage {
+        public static void load(TaskList taskList) {
+            try {
+                String datePattern2 = "MMM-dd-yyyy HH:mm";
```
Perhaps the name of the pattern could give an idea of what the pattern is meant to do?

sopa301 commented on src/main/java/Chatbot.java:
```diff
@@ -0,0 +1,398 @@
+    }
+
+    public static class Ui {
+        public static void startMessage() {
```
Perhaps a more descriptive method name here? I like showStartMessage() since the method has side effects like printing to the screen.

sopa301 reviewed (commented):
Overall, the code looks really neat. However, the whole project being in the same file makes it challenging to read. Consider splitting it into smaller classes if it makes sense.

tanboonkhong commented on src/main/java/Chatbot.java:
```diff
@@ -0,0 +1,398 @@
+        private LocalDateTime startTime;
+        private LocalDateTime endTime;
+
+        String datePattern = "MMM-dd-yyyy HH:mm";
```
Would it be better to change this into a constant and all uppercase if it is intended to be used as a constant?

tanboonkhong commented on src/main/java/Chatbot.java:
```diff
@@ -0,0 +1,398 @@
+            System.out.println(taskList.getTask(taskList.getCount() - 1).getDescription());
+        }
+
+        public static void deadlineMessage(Deadline deadline, TaskList taskList) {
```
could be renamed to displayDeadlineMessage to clarify its purpose

tanboonkhong commented on src/main/java/Chatbot.java:
```diff
@@ -0,0 +1,398 @@
+                } else if (userMessage.substring(0, 5).equalsIgnoreCase("event")) {
+                    try {
+                        int index = userMessage.indexOf("/from");
+                        int index2 = userMessage.indexOf("/to");
```
index and index2 could be renamed to something more descriptive

tanboonkhong commented on src/main/java/Chatbot.java:
```diff
@@ -0,0 +1,398 @@
+        public static void load(TaskList taskList) {
+            try {
+                String datePattern2 = "MMM-dd-yyyy HH:mm";
+                DateTimeFormatter formatter2 = DateTimeFormatter.ofPattern(datePattern2);
```
formatter2 should have a more descriptive name

tanboonkhong reviewed (commented):
There are areas of improvement, to refactoring for better readability and maintainability.

andytoh1 committed:
Merge branch 'add-gradle-support'

# Conflicts:
#	src/main/java/Duke.java

andytoh1 committed:
no message

andytoh1 committed:
no message

andytoh1 committed:
no message

andytoh1 committed:
no message

andytoh1 committed:
no message

andytoh1 committed:
Merge branch 'branch-A-Javadoc'

andytoh1 committed:
no message

andytoh1 committed:
Add assertions

There are several assumptions that are made at any point
within the code. In some cases these will lead to errors.

Documenting these assumptions will lead to easier debugging
and a smoother experience for the user if they can find out more easily
about their wrong inputs.

andytoh1 committed:
Improve code quality

Code quality can contribute greatly to readability of the code.

In this change, multiple long and deeply-nested methods were
broken into smaller components so as to avoid confusion down the line.

andytoh1 committed:
Merge pull request #2 from andytoh1/branch-A-CodeQuality

Improve code quality

andytoh1 committed:
Merge pull request #3 from andytoh1/branch-A-Assertions

Add assertions

andytoh1 committed:
no message

andytoh1 committed:
no message
